src/Model/GroundWaterFlow/gwf3vsc8.f90

```diff
@@ -0,0 +1,1519 @@
+            'UNKNOWN VSC DIMENSION: ', trim(keyword)
+          call store_error(errmsg)
+          call this%parser%StoreErrorUnit()
+        end select
+      end do
+      write (this%iout, '(1x,a)') 'END OF VSC DIMENSIONS'
+    else
+      call store_error('REQUIRED VSC DIMENSIONS BLOCK NOT FOUND.')
+      call this%parser%StoreErrorUnit()
+    end if
+    !
+    ! -- check dimension
+    if (this%nviscspecies < 1) then
+      call store_error('NVISCSPECIES MUST BE GREATER THAN ZERO.')
+      call this%parser%StoreErrorUnit()
+    end if
+    !
+    ! -- return
+    return
+  end subroutine read_dimensions
+
+  !> @ brief Read data for package
+  !!
+  !!  Method to read data for the package.
+  !!
+  !<
+  subroutine read_packagedata(this)
+    ! -- modules
+    ! -- dummy
+    class(GwfVscType) :: this
+    ! -- local
+    character(len=LINELENGTH) :: errmsg
+    character(len=LINELENGTH) :: line
+    integer(I4B) :: ierr
+    integer(I4B) :: iviscspec
+    logical :: isfound, endOfBlock
+    logical :: blockrequired
+    integer(I4B), dimension(:), allocatable :: itemp
+    character(len=10) :: c10
+    character(len=16) :: c16
+    ! -- format
+    character(len=*), parameter :: fmterr = &
+      "('INVALID VALUE FOR IRHOSPEC (',i0,') DETECTED IN VSC PACKAGE. &
+      &IRHOSPEC MUST BE > 0 AND <= NVISCSPECIES, AND DUPLICATE VALUES &
+      &ARE NOT ALLOWED.')"
+! ------------------------------------------------------------------------------
+    !
+    ! -- initialize
+    allocate (itemp(this%nviscspecies))
+    itemp(:) = 0
+    !
+    ! -- get packagedata block
+    blockrequired = .true.
+    call this%parser%GetBlock('PACKAGEDATA', isfound, ierr, &
+                              blockRequired=blockRequired, &
+                              supportOpenClose=.true.)
+    !
+    ! -- parse packagedata block
+    if (isfound) then
+      write (this%iout, '(1x,a)') 'PROCESSING VSC PACKAGEDATA'
+      do
+        call this%parser%GetNextLine(endOfBlock)
+        if (endOfBlock) exit
+        iviscspec = this%parser%GetInteger()
+        if (iviscspec < 1 .or. iviscspec > this%nviscspecies) then
+          write (errmsg, fmterr) iviscspec
+          call store_error(errmsg)
+        end if
+        if (itemp(iviscspec) /= 0) then
+          write (errmsg, fmterr) iviscspec
+          call store_error(errmsg)
+        end if
+        itemp(iviscspec) = 1
+        !
+        this%dviscdc(iviscspec) = this%parser%GetDouble()
+        this%cviscref(iviscspec) = this%parser%GetDouble()
+        call this%parser%GetStringCaps(this%cmodelname(iviscspec))
+        call this%parser%GetStringCaps(this%cauxspeciesname(iviscspec))
+        !
+        if (this%cauxspeciesname(iviscspec) == 'TEMPERATURE') then
+          if (this%idxtmpr > 0) then
+            write (errmsg, '(a)') 'MORE THAN ONE SPECIES IN VSC INPUT IDENTIFIED &
+              &AS "TEMPERATURE".  ONLY ONE SPECIES MAY BE DESIGNATED AS &
+              &TEMPERATURE.'
```

Comment:
we are moving to lowercase messages throughout the mf6 code base.  Maybe take a quick look through all these additions with this in mind.

Reply:
Sounds good.  Messaging switched to lowercase throughout VSC except where variable names, for example `NVISCSPECIES` appear in the msg.  Seems more natural to leave variable names in all caps for easy identification as for alignment with how they appear in mf6io.pdf

Reply:
Addressed in 736b861